nixos/modules/services/web-apps/mediawiki.nix

```diff
@@ -48,10 +48,16 @@ let
     preferLocalBuild = true;
   } ''
     mkdir -p $out/bin
-    for i in changePassword.php createAndPromote.php resetUserEmail.php userOptions.php edit.php nukePage.php update.php; do
-      makeWrapper ${php}/bin/php $out/bin/mediawiki-$(basename $i .php) \
-        --set MEDIAWIKI_CONFIG ${mediawikiConfig} \
-        --add-flags ${pkg}/share/mediawiki/maintenance/$i
+    for i in changePassword createAndPromote deleteUserEmail resetUserEmail userOptions edit nukePage update; do
+      echo "#! ${pkgs.runtimeShell}
+      become=(exec)
+      if [[ \"\$(id -u)\" != ${user} ]]; then
+        become=(exec /run/wrappers/bin/sudo -u ${user} --)
+      fi
+      \"\''${become[@]}\" ${php}/bin/php ${pkg}/share/mediawiki/maintenance/run.php --conf ${mediawikiConfig} \
+        ${pkg}/share/mediawiki/maintenance/$i.php \"\$@\"
+      " > $out/bin/mediawiki-$i
+      chmod +x $out/bin/mediawiki-$i
```

Comment:
- rather than echoing this multiline string, heredoc i.e << EOF can be used, makes it far more readable.
  eg. https://github.com/numtide/build-go-cache/pull/12/files#diff-95b29ca6e6d040f04802e06f770b1c4d7235a72edb5bc6ddaecaa425a32f5672R68
  also using 'EOF' to avoid escaping quotes, eg. https://github.com/numtide/build-go-cache/pull/12#discussion_r1777777221
- `${"$"}` to escape dollar, [source](https://discourse.nixos.org/t/how-to-escape-in-indented-string/41114/2)
- `sed` to remove leading indentation in the resulting output script. and EOF indentation should be left like this, same level as for.
- added `importDump` to the scripts
  - mediawiki-importDump failed with the `--conf ${}` flag, but works with `MEDIAWIKI_CONFIG` env var, so using makeWrapper.
- added `run` to the scripts, allows using `mediawiki-run <importDump/Version/initSiteStats/rebuildrecentchanges/...>` i.e. relative script name inside maintenance folder.
  - also `mediawiki-maintenance` which needs absolute path to maintenance scripts and needs to be run with `runuser/su mediawiki`.
  - there is an executable `run` in maintenance dir, which is not used, I wrap run.php. 

here's the patch

```diff
diff --git a/nixos/modules/services/web-apps/mediawiki.nix b/nixos/modules/services/web-apps/mediawiki.nix
index f78880be6549..586ca1768e25 100644
--- a/nixos/modules/services/web-apps/mediawiki.nix
+++ b/nixos/modules/services/web-apps/mediawiki.nix
@@ -48,16 +48,25 @@ let
     preferLocalBuild = true;
   } ''
     mkdir -p $out/bin
-    for i in changePassword createAndPromote deleteUserEmail resetUserEmail userOptions edit nukePage update; do
-      echo "#! ${pkgs.runtimeShell}
+    makeWrapper ${php}/bin/php $out/bin/mediawiki-maintenance \
+      --set MEDIAWIKI_CONFIG ${mediawikiConfig} \
+      --add-flags ${pkg}/share/mediawiki/maintenance/run.php
+    for i in changePassword createAndPromote deleteUserEmail resetUserEmail userOptions edit nukePage update importDump run; do
+      script="$out/bin/mediawiki-$i"
+    sed -E 's/^ {2}//' <<'EOF' >"$script"
+      #!${pkgs.runtimeShell}
       become=(exec)
-      if [[ \"\$(id -u)\" != ${user} ]]; then
+      if [[ "$(id -u)" != ${user} ]]; then
         become=(exec /run/wrappers/bin/sudo -u ${user} --)
       fi
-      \"\''${become[@]}\" ${php}/bin/php ${pkg}/share/mediawiki/maintenance/run.php --conf ${mediawikiConfig} \
-        ${pkg}/share/mediawiki/maintenance/$i.php \"\$@\"
-      " > $out/bin/mediawiki-$i
-      chmod +x $out/bin/mediawiki-$i
+      "${"$"}{become[@]}" ${placeholder "out"}/bin/mediawiki-maintenance \
+    EOF
+      if [[ "$i" != "run" ]]; then
+        echo "  ${pkg}/share/mediawiki/maintenance/$i.php \"\$@\"" >>"$script"
+      else
+        echo "  ${pkg}/share/mediawiki/maintenance/\$1.php \"\''${@:2}\"" >>"$script"
+      fi
+      chmod +x "$script"
     done
   '';
```

Reply:
Thanks!

> `sed` to remove leading indentation in the resulting output script. and EOF indentation should be left like this, same level as for.

Or we just use cat and remove 2 indentation. I don't think it is important to have the two spaces there.